build/scripts/bootstrap.sh

```diff
@@ -125,82 +111,190 @@ function installBuildDependencies
 function installDeveloperTools
 {
     echo "[INFO] Installing developer tools"
-    if [[ "$PLATFORM" == "osx" ]]
+    if [ "$is_macos" = true ];
     then
         brew install cpplint
-    elif isSupportedLinux
+    elif [ "$is_linux" = true ];
     then
         apt-get install -y python-pip
         pip install cpplint
 
         # Installs to a non-standard location so add to PATH manually
         export PATH=$PATH:~/.local/bin
     else
-        "[INFO] Developer tools not supported on this platform"
+        echo "[INFO] Developer tools not supported on this platform"
     fi
 }
 
 function installContainerTools
 {
-    if !isSupportedLinux
+    if [ "$is_linux" = true ];
     then
-        echo "[INFO] Container builds not supported on this platform"
-    else
         apt-get install -y curl
 
         echo "[INFO] Installing Docker"
         # Install docker to enable building cross-arch for arm
         # Instructions located at: https://docs.docker.com/engine/install/ubuntu/
         curl -fsSL https://get.docker.com -o get-docker.sh
         sh get-docker.sh
+    else
+        echo "[INFO] Container builds not supported on this platform"
     fi
 }
 
 function installQemu
 {
-    if !isSupportedLinux
+    if [ "$is_linux" = true ];
     then
-        echo "[INFO] Emulated builds not supported on this platform"
-    else
         echo "[INFO] Installing Qemu for cross-arch support"
         # Install qemu for cross-arch support
         apt-get -y install qemu binfmt-support qemu-user-static
 
         # Register qemu with docker to more easily run cross-arch containers
         docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+    else
+        echo "[INFO] Emulated builds not supported on this platform"
     fi
 }
 
 function installAll
 {
     echo "Setting up development environment for do-client"
-
     installBuildDependencies
-
-    if isSupportedLinux
-    then
-        installDeveloperTools
-        installContainerTools
-        installQemu
-    fi
+    installDeveloperTools
+    installContainerTools
+    installQemu
 }
 
 function isSupportedLinux()
 {
-    if [[ "$PLATFORM" == "debian10" || "$PLATFORM" == "ubuntu1804" || "$PLATFORM" == "ubuntu2004" ]];
+    if [[ ("$OS" == "ubuntu" && ($VER == "18.04" || $VER == "20.04"))
+        || ("$OS" == "debian" && ($VER == "10" || $VER == "11")) ]];
     then
         return 0
     else
         return 1
     fi
 }
 
+function isSupportedMacOS()
+{
+    # No specific version check needed at present
+    return 0
+}
+
+# From DU project: https://github.com/Azure/iot-hub-device-update/blob/main/scripts/install-deps.sh
+function determine_machine_architecture()
+{
+    local arch=''
+    arch="$(uname -m)"
+    local ret_val=$?
+    if [[ $ret_val != 0 ]]; then
+        echo "Failed to get cpu architecture."
+        return 1
+    else
+        if [[ $arch == aarch64* || $arch == armv8* ]]; then
```

Comment:
see my other comment -- these are mutually exclusive.  e.g. $ARCH_FLAVOR="arm32" would be better IMHO

Reply:
Yes, good call.